openfisca_france/model/prestations/minima_sociaux/asi_aspa.py

```diff
@@ -204,6 +204,43 @@ class asi(Variable):
     set_input = set_input_divide_by_period
     calculate_output = calculate_output_add
 
+    def formula_2020_04_01(individu, period, parameters):
+        maries = individu.famille('maries', period)
+        en_couple = individu.famille('en_couple', period)
+        asi_aspa_nb_alloc = individu.famille('asi_aspa_nb_alloc', period)
+        base_ressources = individu.famille('asi_aspa_base_ressources', period)
+        P = parameters(period).prestations.minima_sociaux
+
+        demandeur_eligible_asi = individu.famille.demandeur('asi_eligibilite', period)
+        demandeur_eligible_aspa = individu.famille.demandeur('aspa_eligibilite', period)
+        conjoint_eligible_asi = individu.famille.conjoint('asi_eligibilite', period)
+        conjoint_eligible_aspa = individu.famille.conjoint('aspa_eligibilite', period)
+
+        # Un seul éligible
+        elig1 = ((asi_aspa_nb_alloc == 1) & (demandeur_eligible_asi | conjoint_eligible_asi))
+        # Couple d'éligibles mariés
+        elig2 = demandeur_eligible_asi & conjoint_eligible_asi & maries
+        # Couple d'éligibles non mariés
+        elig3 = demandeur_eligible_asi & conjoint_eligible_asi & not_(maries)
+        # Un seul éligible et époux éligible ASPA
+        elig4 = ((demandeur_eligible_asi & conjoint_eligible_aspa) | (conjoint_eligible_asi & demandeur_eligible_aspa)) & maries
+        # Un seul éligible et conjoint non marié éligible ASPA
+        elig5 = ((demandeur_eligible_asi & conjoint_eligible_aspa) | (conjoint_eligible_asi & demandeur_eligible_aspa)) & not_(maries)
+
+        plafond_ressources = (
+            elig1 * (P.asi.plafond_ressource_seul * not_(en_couple) + P.asi.plafond_ressource_couple * en_couple)
+            + elig2 * P.asi.plafond_ressource_couple
+            + elig3 * P.asi.plafond_ressource_couple
+            + elig4 * P.aspa.plafond_ressources_couple
+            + elig5 * P.aspa.plafond_ressources_couple) / 12
+
+        montant_servi_asi = max_(plafond_ressources - base_ressources, 0)
+
+        return montant_servi_asi * (
+            + individu.has_role(Famille.DEMANDEUR) * demandeur_eligible_asi * (elig1 + elig2 / 2 + elig3 / 2)
+            + individu.has_role(Famille.CONJOINT) * conjoint_eligible_asi * (elig1 + elig2 / 2 + elig3 / 2)
```

Comment:
Si j'ai bien compris, dans cette formule, les couples dans les cas `elig4` et `elig5` n'ont jamais le droit à l'ASI ?
Donc si votre conjoint/partenaire touche l'ASPA, cela enlève votre droit à l'ASI ?

Reply:
Bonjour @Sasha-Laniece !

nous avons bien noté vos questions pour les poser à un expert.
Pour les couples, si un conjoint/partenaire touche l'ASPA, cela n'enlève pas le droit à l'ASI par contre ce n'est pas le même plafond des ressources appliqué.

Reply:
Super ! Je vous ai fait le rebase suite à l'harmonisation. Hâte d'avoir votre retour et les tests pour merger cette PR :)